app/models/manageiq/providers/redhat/infra_manager/vm_import.rb

```diff
@@ -0,0 +1,85 @@
+module ManageIQ::Providers::Redhat::InfraManager::VmImport
```

Comment:
This is just a thoughton making things more open-close and have less responsibilities.
As far as I understand this class currently does three things, it validates that the import is possible, it then builds the import params, then it does the import.
If we want it to be generic I think it should implement the strategy pattern, there should be some kind of an Importer which can be VmwImporter or SomethingelseImporter, there should be a params builder.
I get the feeling that the dupication between lines 22-29 and 48-56 means something can be improved (Its not exact duplication, but it feels like I would like to do `params.merge(provider: OvirtSDK4::ExternalVmProviderType::VMWARE)` instead of all of that. So I would have a ParamsBuilder and pass what it returns to the Importer (both strategies selected according to the Vm provider type), then you could implement each of them for any of the providers in the future.
I did not actually try to implement this so maybe its not a good idea, but that is what I would try.